### ISSUE TITLE
js_index.loads fails to load search index

### ISSUE THREAD

**Issue by lazka** [sphinx-doc/sphinx#2902]
I was using sphinx.search.js_index.loads to do some post processing on the searchindex and since some update it no longer works.

It comes down to the following failing to parse:

```
from sphinx.search import js_index
js_index.loads('Search.setIndex({envversion:49,terms:{_new:21})')
```

**Comment by Blendify** [comment]
Does sphinx 1.4.4 work?

**Comment by lazka** [comment]
No, same error. Also with 1.3 and 1.2. I guess the problem comes from the search index now containing variables starting with "_"

**Comment by lazka** [comment]
Yeah. My older searchindex files produced with older versions contain `"_new":21` for the same docs

**Comment by lazka** [comment]
```
>>> from sphinx.util.jsdump import dumps, loads
>>> loads(dumps({"_foo": 42}))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sphinx/util/jsdump.py", line 187, in loads
    raise ValueError("read error at pos %d" % i)
ValueError: read error at pos 1
```

**Comment by tk0miya** [comment]
Sorry for inconvenience. It seems a bug since Sphinx-1.4.5.

```
>>> import sphinx
>>> sphinx.version_info
(1, 4, 4, 'final', 0)
>>> from sphinx.util.jsdump import dumps, loads
>>> loads(dumps({"_foo": 42}))
{'_foo': 42}
```

```
>>> import sphinx
>>> sphinx.version_info
(1, 4, 5, 'final', 0)
>>> from sphinx.util.jsdump import dumps, loads
>>> loads(dumps({"_foo": 42}))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sphinx/util/jsdump.py", line 187, in loads
    raise ValueError("read error at pos %d" % i)
ValueError: read error at pos 1
```

I'll take a look later.

**Comment by tk0miya** [comment]
Fixed at acd4d9f.
Thank you for reporting!

**Comment by lazka** [comment]
Thanks!
